packages/repository-json-schema/src/keys.ts

```diff
@@ -6,6 +6,10 @@
 import {MetadataAccessor} from '@loopback/metadata';
 import {JSONSchema6 as JSONSchema} from 'json-schema';
 
+/**
+ * TODO(semver-major) remove these constants in the next major version
+ * @deprecated
+ */
```

Comment:
Can I ask why they're being removed?

Reply:
Of course you can! 😄 

Cross-posting my comment from the discussion thread https://github.com/strongloop/loopback-next/pull/3199#discussion_r296590295:

> In the future, we will be adding more schema options like `exclude` (see https://github.com/strongloop/loopback-next/issues/2653). Because options can be combined, e.g. `partial` with `exclude`, the number of possible combinations (and thus cache) keys is growing fast. I find it impractical to enumerate all valid keys in a TypeScript enum type. We will need 4 entries now, 8 entries when we implement `exclude`, 16 entries when we introduce another option, etc.
>
> That's why I switched the indexer type from `options in MODLE_TYPE_KEYS` to `key: string`.

I am going to improve this comment to tell developers to use `buildModelCacheKey` instead.